### ISSUE TITLE
images filter 'dangling' does not filter images, always returns all images

### ISSUE THREAD

**Issue by morxa** [docker/swarm#1700]
In the following: `alias swarm=docker -H myswarm:2375`

Description of problem:
`swarm images -f dangling=true` does not filter the images, all images are marked as dangling. The same holds for `dangling=false`.

`docker version` on the client:

```
Client:
 Version:      1.9.1-cs3
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   28d463f
 Built:        Mon Jan  4 22:11:37 UTC 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1-cs3
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   28d463f
 Built:        Mon Jan  4
```

`docker version` on the host running the swarm:

```
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64
```

Output of `swarm info`:

```
Containers: 6
Images: 14
Role: primary
Strategy: spread
Filters: health, port, dependency, affinity, constraint
Nodes: 2
 ad-sim01: 53.255.82.41:2375
  └ Status: Healthy
  └ Containers: 3
  └ Reserved CPUs: 4 / 16
  └ Reserved Memory: 7 GiB / 32.18 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.19.0-25-generic, operatingsystem=Ubuntu 14.04.3 LTS, storagedriver=aufs
 ad-sim02: 53.255.82.52:2375
  └ Status: Healthy
  └ Containers: 3
  └ Reserved CPUs: 4 / 16
  └ Reserved Memory: 7 GiB / 32.99 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.19.0-25-generic, operatingsystem=Ubuntu 14.04.3 LTS, storagedriver=aufs
CPUs: 32
Total Memory: 65.17 GiB
Name: 51a1d290da7a
```

`uname -a` on the swarm host:

```
Linux ci 3.16.0-30-generic #40~14.04.1-Ubuntu SMP Thu Jan 15 17:43:14 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

How reproducible: always

How to reproduce:
1. Pull any image: `swarm pull ubuntu:14.04`
2. Run `swarm images -f dangling=true`

Actual results:

```
% swarm images -f dangling=true | grep ubuntu
ubuntu                                 14.04               6cc0fc2a5ee3        8 days ago          187.9 MB
```

Expected results:
The resulting list should not show the ubuntu image.

Additional info:
If not running on swarm, the filter works as expected:

```
% docker pull ubuntu:14.04
% docker images -f dangling=true | grep ubuntu ; echo $?
1
```

**Comment by dongluochen** [comment]
Docker `dangling` filter was added recently and swarm has not supported it yet.

**Comment by vieux** [comment]
@dongluochen recently ? I believe it was introduced in engine 1.2.0 :)

**Comment by dongluochen** [comment]
@vieux I didn't check all the code history :(

**Comment by jimmyxian** [comment]
Swarm does not support `dangling`
Also, I think difficult to support. 
Image data which doesn't have repotag in swarm have two kinds of classification( `danging` and `intermediate images` ), and we can not distinguish.

Also, `docker images` in swarm does not show the image which doesn't have repotag by default.

**Comment by schmunk42** [comment]
Swarm should at least throw a warning in this case - I just wiped out all my swarm images...

**Comment by morxa** [comment]
Seems to be fixed in #2670.
